### PR TITLE
ref(metrics-indexer): Use commit policy instead of hand-written throttling [sns-1862]

### PR DESCRIPTION
**Pull request by untitaker**
Adopt new goodness from arroyo by using its commit policy feature to
throttle commits instead of hand-rolled code.

### CODE REVIEW

File: src/sentry/sentry_metrics/consumers/indexer/multiprocess.py [hunk 1]
```diff
@@ -221,12 +186,7 @@ def join(self, timeout: Optional[float] = None) -> None:
                 timeout = 5.0
             self.__producer.flush(timeout)
 
-        if self.__callbacks:
-            logger.info(f"Committing {self.__callbacks} messages...")
-            self.__commit_function(self.__produced_message_offsets)
-            self.__callbacks = 0
-            self.__produced_message_offsets = {}
```
Comment (untitaker) [inline src/sentry/sentry_metrics/consumers/indexer/multiprocess.py:228]: We are now unconditionally clearing this map even if the commitpolicy throttled the commit call. I hope that is okay. I was initially not entirely sure what would happen if one would do this:

1. call `commit({1: 2})`, which gets throttled
2. call `(commit({2: 4})`, which gets accepted

will partition 1 be committed?

the answer seems to be yes if I read https://github.com/getsentry/arroyo/blob/eadfaa1436ac70dbeaed914737acf854ef76a5b0/arroyo/processing/processor.py#L151-L153 correctly.

**Review by nikhars — APPROVED** [review]

File: src/sentry/sentry_metrics/consumers/indexer/multiprocess.py (hunk 1 above)
Comment (lynnagara) [reply src/sentry/sentry_metrics/consumers/indexer/multiprocess.py:228]: Yup it will. All changes are staged even if throttled for commit later.

**Review by lynnagara — APPROVED** [review]

**Comment by untitaker** [comment]
as suggested by filippo i'll run a load test on this tomorrow before merging

**Comment by lynnagara** [comment]
Fwiw this change (just the commit policies) is unlikely to have performance impacts. It's more switching to the ProduceStep and the KafkaProducer that would need to be tested. Still no objection if you wanted to run the test anyway.

**Comment by untitaker** [comment]
https://github.com/getsentry/arroyo/pull/155

File: requirements-base.txt [hunk 2]
```diff
@@ -54,7 +54,7 @@ requests>=2.25.1
 rfc3339-validator>=0.1.2
 rfc3986-validator>=0.1.1
 # [end] jsonschema format validators
-sentry-arroyo>=2.0.0
+https://github.com/getsentry/arroyo/archive/06df377477f0a1de7844f1fac9d275b2822127c7.zip
```
Comment (evanh) [inline requirements-base.txt:57]: Why these changes?
